services:
  db:
    image: postgres
    volumes:
      - ./data/db:/var/lib/postgresql/data
    env_file: .env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 2s
      timeout: 5s
      retries: 5
    networks:
        - codata_dw
  web:
    build: .
    stdin_open: true 
    tty: true  
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
    ports:
      - "8000:8000"
    env_file: .env
    depends_on:
      db:
        condition: service_healthy
    networks:
        - codata_dw

  rabbitmq:
    image: rabbitmq:3.11-management-alpine
    hostname: rabbitmq
    env_file: .env
    ports:
      - "5672"
      - "15672"
    networks:
        - codata_dw
    healthcheck:
      test: rabbitmq-diagnostics -q ping
      interval: 30s
      timeout: 30s
      retries: 3

  worker:
    build: .
    stdin_open: true 
    tty: true  
    command: celery -A codata_dw worker -l INFO
    volumes:
      - .:/code
    env_file: .env
    depends_on:
      db:
        condition: service_healthy
      rabbitmq:
        condition: service_healthy
    networks:
        - codata_dw

networks:
  codata_dw:
    driver: bridge